src/libs/es_writer.py

```diff
@@ -100,3 +91,22 @@ def create_index(self, index_name):
             msg = "Exception encountered during executing ESWriter.create_index()"
             # Log the full stack trace, prepend a line with our message
             logger.exception(msg)
+
+
+def _get_configuration(index_name):
+    if index_name == 'portal':
```

Comment:
Currently the `indexer.py` is probably the only place that uses the `es_writer.py` and when it creates the indices, there's no such index named `portal`. The actual ones are `hm_public_portal` and `hm_consortium_portal`, but from the end user perspective, they'll see `portal` as the index name because we are not exposing the actual real index names to the end users.

Was your intention to use the mapping from `config.yaml` when  `hm_public_portal` and `hm_consortium_portal` are being created?

Reply:
My understanding of the `config.yaml` usage is for creating the portal indices, if this is correct, I'm also thinking that we should move this loading of the mapping details to the `indexer.py`. Because the `indexer.py` reads in the indices configuration from https://github.com/hubmapconsortium/search-api/blob/test-release/src/instance/app.cfg.example#L29 and creates the indices at https://github.com/hubmapconsortium/search-api/blob/test-release/src/elasticsearch/indexer.py#L73. The `es_writer.py` was designed as a library and it shouldn't rely on reading a `config.yaml` for creating certain indices. 

As a result, we can add another parameter to the `ESWriter.create_index(index_name, config)` and the `config` would include the settings and mappings definition, which can be passed in at runtime. Eventually, we can specify the settings and mappings in `indexer.py` for both portal indices and the other original indices. If someone else wants to use the `es_writer.py`, they'll be able to pass in their own configurations for creating indices easily without making changes to the `es_writer.py`. 

@shirey @mccalluc does this sound like a good move?

Reply:
@mccalluc @yuanzhou and I just discussed this, what Zhou is saying seems correct- these configuration changes only need to happen when the indices are created.  Our indices are currently created when we kick off a full-reindex with a standalone script, but we could separate that piece if helpful to you during development.

Reply:
> Was your intention to use the mapping from config.yaml when hm_public_portal and hm_consortium_portal are being created?

Assuming these are the instances that have been run through `src/elasticsearch/addl_index_transformations/portal`, yes.

> As a result, we can add another parameter to the ESWriter.create_index(index_name, config) and the config would include the settings and mappings definition, which can be passed in at runtime. ... does this sound like a good move?

As far as I follow it, yes.

> these configuration changes only need to happen when the indices are created.

yes.

>  Our indices are currently created when we kick off a full-reindex with a standalone script, but we could separate that piece if helpful to you during development.

I'm not sure that that is necessary: I think the multiple iterations needed for the recent nested-metadata PR were exceptional.

@shirey @yuanzhou : Here are the next steps as I see them:
- Chuck will make a new PR, branching from just before the last commit.
- Zhou will review, and if appropriate, merge.
- Zhou or Bill will do what is needed so that the mappings Chuck specifies are applied when a new index is set up.
- Chuck will have occasional mapping updates, and will be able confirm some aspects of the functionality with `test-es.py`... but other issues may not be apparent without pointing a portal instance at the result of a reindex... Using TEST or DEV for this works for me... but it might not work for you, in which case there should be guidance on setting up a local instance.
- At some point, the PROD index is updated. I believe a microservices architecture should mean that extended downtime is not necessary... but I don't think that's the case here.